lib/LaTeXML/Core/Rewrite.pm

```diff
@@ -348,20 +470,40 @@ sub domToXPath_rec {
         my $pred = LaTeXML::Common::Font::font_match_xpaths($font);
         push(@predicates, $pred); } }
 
-    return $qname . "[" . join(' and ', grep { $_ } @predicates, @extra_predicates) . "]"; }
+    if ($pos) {
+      unshift(@predicates, 'self::' . $qname);
+      $qname = '*';
+      unshift(@predicates, 'position()=' . $pos); }
+    my $preds = join(' and ', grep { $_ } @predicates);
+    return ($axis . '::' . $qname . ($preds ? "[" . $preds . "]" : ''), 1, 0, @wilds);
+  }
 
   elsif ($type == XML_TEXT_NODE) {
-###    "text()='".$node->textContent."'"; }}
-    return "*[text()='" . $node->textContent . "']"; } }
+    return ("*[text()='" . $node->textContent . "']", 1, 0); } }
```

Comment:
You probably want to escape any apostrophes in the returned `->textContent` to avoid malformed XPaths, though I can see how that's a rare edge case.

Reply:
you might be right, but I can't figure out how to get a plain apostrophe in there!!

Reply:
Oh, ok. This TeX doc will give you a Fatal error:
```tex
\documentclass{article}
\usepackage{latexml}
\usepackage{textcomp}
\begin{document}

\lxDeclare[role=ID]{$x\textquotesingle$}
\end{document}
```

with:
```

Fatal:perl:die Perl died
	at String; line 0 col 0 - line 0 col 0
	XPath error : Invalid predicate
	descendant-or-self::ltx:XMTok[@role='UNKNOWN' and text()='x' and @_font and contains(@_font,'[math,') and contains(@_font,',medium,') and contains(@_font,',italic,')][following-sibling::*[position()=1 and self::ltx:XMTok and @role='UNKNOWN' and text()=''' and @_font and contains(@_font,'[serif,') and contains(@_font,',medium,') and contains(@_font,',upright,')]][@_pvis and @_cvis]
	                                                                                                                                                                                                                                                              ^
	In Core::Document[@0x56466e6558c8] undef
1 fatal error at /home/deyan/perl5/lib/perl5/LaTeXML/Common/XML/XPath.pm line 36.
```

Reply:
Ah, very clever.  OK, try this patch.